projects/addon-mobile/components/swipe-action/swipe-actions-auto-close.directive.ts

```diff
@@ -0,0 +1,31 @@
+import {Directive, ElementRef, Inject, Input} from '@angular/core';
+import {tuiGetActualTarget} from '@taiga-ui/cdk';
+
+@Directive({
+    standalone: true,
+    selector: 'tui-swipe-actions[autoClose]',
+    host: {
+        '(document:pointerdown.silent)': 'onPointer($event)',
+    },
+})
+export class TuiSwipeActionsAutoCloseDirective {
+    @Input()
+    autoClose = true;
+
+    constructor(@Inject(ElementRef) private readonly el: ElementRef<HTMLElement>) {}
```

Comment:
`inject`? We can then store `nativeElement` itself in `el` and make code shorter, maybe even get rid of `close` method?